### PR TITLE
[#77] Binding Question & Answer flow

### LINKED ISSUE #77: [Integrate] As a user, I can select an answer and move forward
## Why

The user should be able to select an Answer option, then move forward. 

In the end, they can complete the Survey.
 
## Acceptance Criteria
- Allow the user to select an answer
- Allow the user to move back and forth, we need to persist the choice that the users had chosen.
- Make sure that all the choices are ready to be submitted in the end.

### PR DESCRIPTION
**Pull request by sleepylee**
Closes #77 

## What happened 👀

Finally binding data and allow the user to navigate through all the question and see all the answers.
 
## Insight 📝

- Binding data from the question (title) to the Answer with the corresponding answer type.
- Some minor UI update because it's so ugly and hard to see the content right now: adding dimmed overlay; adjust text length limit to avoid breaking UI or collapse fields
- Very small refactor at the main `Stack` to make it easier to review.
 
## Proof Of Work 📹

https://user-images.githubusercontent.com/13435717/125808218-bec92c05-d9e2-48c8-85c6-e1b03c275619.mov

### CODE REVIEW

**Review by markgravity — APPROVED** [review]
LGTM 🤖

File: lib/pages/survey/survey_controller.dart [hunk 1]
```diff
@@ -10,6 +10,17 @@ class SurveyController extends GetxController {
...
+  int get questionAmount => optionalSurvey.value.questions.length;
+
+  final _currentQuestion = 0.obs;
+
+  int get currentQuestionIndex => _currentQuestion.value;
+
+  SurveyQuestion get currentQuestion =>
+      optionalSurvey.value.questions[currentQuestionIndex];
+
+  String get indexString => _getIndexString();
```
Comment (manh-t) [inline lib/pages/survey/survey_controller.dart:22]: perhaps `questionPositionString` 💭

File: lib/pages/survey/survey_question_page.dart [hunk 2]
```diff
@@ -2,111 +2,147 @@ import 'package:cached_network_image/cached_network_image.dart';
...
-              ),
-            ),
-            SafeArea(
-              child: Column(
-                crossAxisAlignment: CrossAxisAlignment.start,
+      body: GetBuilder(
+          init: SurveyController(),
+          builder: (_) {
+            return GetX<SurveyController>(builder: (controller) {
+              //return Text(controller.indexString);
```
Comment (manh-t) [inline lib/pages/survey/survey_question_page.dart:20]: can we remove this comment? 😅

File: lib/pages/survey/survey_question_page.dart [hunk 3]
```diff
@@ -2,111 +2,147 @@ import 'package:cached_network_image/cached_network_image.dart';
...
+        onPressed: () {
+          // TODO: update this behavior: show confirmation dialog
+          Get.back();
+        },
+        shape: CircleBorder(),
+      ),
+    );
+  }
+
+  Widget _buildTitle(BuildContext context, String indexString, String text) {
```
Comment (manh-t) [inline lib/pages/survey/survey_question_page.dart:74]: same here `questionPositionString` 👍

**Review by manh-t — COMMENTED** [review]
@sleepylee I'm just curious is there another part for this task? I don't see the step to add answer into a form 😄

File: lib/pages/survey/survey_controller.dart (hunk 1 above)
Comment (sleepylee) [reply lib/pages/survey/survey_controller.dart:22]: maybe `indexTitleText` could be a clearer name 🤔

**Comment by sleepylee** [comment]
@manh-t 
>is there another part for this task? I don't see the step to add answer into a form
yes - form submission, data gathering will be on another PR. I make every PR super clean & small to review 😎

File: lib/pages/survey/survey_question_page.dart (hunk 2 above)
Comment (sleepylee) [reply lib/pages/survey/survey_question_page.dart:20]: 🤪

File: lib/pages/survey/survey_controller.dart (hunk 1 above)
Comment (manh-t) [reply lib/pages/survey/survey_controller.dart:22]: that works for me 🙆‍♂️

File: lib/pages/survey/survey_question_page.dart (hunk 3 above)
Comment (sleepylee) [reply lib/pages/survey/survey_question_page.dart:74]: I make this becomes:

`Widget _buildTitle(BuildContext context, String indexTitle, String questionTitle)`

so I think this is clearer

**Review by manh-t — APPROVED** [review]
Looks good now 🙏

**Pull request merged**